src/qml/modules/Shotcut/Controls/GradientMapControl.qml

```diff
@@ -0,0 +1,307 @@
+            for (var idx = 0; idx < _stopHandles.length; idx++) {
+                _stopHandles[idx].destroy();
+            }
+            _stopHandles = [];
+            return;
+        }
+        stops.sort(_compareStop);
+        var newStops = Array.from(stops, function (stop) { return stopComponent.createObject(gradientView, stop); });
+        gradientView.stops = newStops;
+        for (var idx = 0; idx < _stopHandles.length; idx++) {
+            _stopHandles[idx].destroy();
+        }
+        var newHandles = [];
+        for (var idx = 0; idx < stops.length; idx++) {
+            newHandles.push(stopHandle.createObject(gradientFrame, {
+                "stopIndex": idx,
+                "stopList": stops,
+                "prev": idx == 0 ? null : stops[idx - 1],
+                "next": idx == stops.length - 1 ? null : stops[idx + 1]
+            }));
+        }
+        _stopHandles = newHandles;
+    }
+
+    function _setStopColor(index, color, position) {
+        stops[index].color = color;
+        stops[index].position = position;
+        if (stopSpinner.value !== index) {
+            stopSpinner.value = index;
+        }
+        _updateColorDisplay();
+        gradientChanged();
+    }
+
+    function _setStopPosition(index, position) {
+        gradientView.stops[index].position = position;
+    }
+
+    function _insertStop(index, stop) {
+        var newstops = Array.from(stops);
+        newstops.splice(index, 0, stop);
+        stops = newstops;
+        _updateColorDisplay();
+        gradientChanged();
+    }
+
+    function _removeStop(index) {
+        var newstops = Array.from(stops);
+        newstops.splice(index, 1);
+        stops = newstops;
+        _updateColorDisplay();
+        gradientChanged();
+    }
+
+    Component.onCompleted: _updateColorDisplay()
+    onStopsChanged: _updateColorDisplay()
+
+    Button {
+        enabled: stops.length <= 10
+        icon.name: "list-add"
+        onClicked: {
+            if (stops.length < 1) {
+                _insertStop(0, {"color": "#ff000000", "position": 0.5});
+                return;
+            }
+            var index = stopSpinner.value;
+            var prev = {};
+            var next = {};
+            if (index === 0) {
+                prev = stops[index];
+                if (prev.position <= 0.0) {
+                    ++index;
+                    next = stops[index];
+                } else {
+                    _insertStop(0, {"color": prev.color, "position": 0.0});
+                    return;
+                }
+            } else  {
+                prev = stops[index - 1];
+                next = stops[index];
+            }
+            if (typeof next === 'undefined') {
+                _insertStop(index, {"color": prev.color, "position": 1.0});
+                return;
+            }
+            _insertStop(index, {"color": next.color, "position": (prev.position + next.position) / 2.0});
+        }
+    }
+
+    Button {
```

Comment:
The + and - buttons are too wide. Try adding to them `implicitWidth: height`

Alternatively, you could make this more consistent with our other gradient control by removing these buttons and placing a spinner with the number of stops after the control on the same row. See filter "Audio Spectrum Visualization"
![image](https://github.com/user-attachments/assets/f91467de-08da-4be4-8bb0-350668291d14)
(Except Gradient Map filter could use "stops" instead of "colors" as the suffix.)

Reply:
The add and remove buttons at least gives me the chance to select where to insert a stop and which one to delete, controlling a number of stops only makes sense for me if they are equidistant.

Reply:
OK